c++/src/Timezone.cc

```diff
@@ -659,7 +659,15 @@ namespace orc {
   const char* getTimezoneDirectory() {
     const char* dir = getenv("TZDIR");
     if (!dir) {
-      dir = DEFAULT_TZDIR;
+      // this is present if we're in an activated conda environment
+      const char* conda_prefix = getenv("CONDA_PREFIX");
+      if (conda_prefix) {
+        std::string conda_dir(conda_prefix);
+        conda_dir += "/share/zoneinfo";
+        dir = conda_dir.c_str();
```

Comment:
This unfortunately returns a dangling pointer.

Reply:
Not sure I understand (but then, my C++ is limited...); `dir` is `const char *`, and `std::string` doesn't auto-convert. How should I set `dir` with the correct value? I can also construct the `conda_dir` path differently if you want, I just followed the example from `getTimezoneByName` that was using `std::string`.

Reply:
c_str() returns the pointer in the backing string object which is a temporary object and is destructed when the function returns.

Reply:
It seems we need to change the function to return std::string instead

Reply:
> It seems we need to change the function to return std::string instead

I tried that now; not sure how you'd like to do this stylistically (i.e. the early return for `condaDir`); for now I didn't want to touch the `getenv` bits, because that always returns `char *`.